docs/BOOT_LOADER_SPECIFICATION.md

```diff
@@ -7,120 +7,63 @@ SPDX-License-Identifier: LGPL-2.1-or-later
+  installing or updating kernel packages
+* UI developers, to implement user interfaces that list and select among the
+  available boot options
 
-## Why not simply rely on the EFI boot menu logic?
+## The boot partition
 
-EFI is not ubiquitous, especially not in embedded systems. If you have an EFI
-system, it provides a boot options logic that can offer similar
-functionality. Here's why we think that it is not enough for our uses:
+Everything described below is located on a file system that will be called
+`$BOOT`. The boot loader or user-space programs reading the boot loader
+configuration should pick `$BOOT` according to the following rules:
 
-* The various EFI implementations implement the boot order/boot item logic to
-  different levels. Some firmware implementations do not offer a boot menu at
-  all and instead unconditionally follow the EFI boot order, booting the first
-  item that is working.
-* If the firmware setup is used to reset all data usually all EFI boot entries
-  are lost, making the system entirely unbootable, as the firmware setups
-  generally do not offer a UI to define additional boot items. By placing the
-  menu item information on disk, it is always available, regardless if the BIOS
-  setup data is lost.
-* Harddisk images should be movable between machines and be bootable without
-  requiring explicit EFI variables to be set. This also requires that the list
-  of boot options is defined on disk, and not in EFI variables alone.
-* EFI is not universal yet (especially on non-x86 platforms), this
-  specification is useful both for EFI and non-EFI boot loaders.
-* Many EFI systems disable USB support during early boot to optimize boot
-  times, thus making keyboard input unavailable in the EFI menu. It is thus
-  useful if the OS UI has a standardized way to discover available boot options
-  which can be booted to.
-
-## Technical Details
+* On disks with an MBR partition table:
 
-Everything described below is located on a placeholder file system `$BOOT`. The
-installer program should pick `$BOOT` according to the following rules:
+  * A partition with the type ID of 0xEA shall be used as `$BOOT`.
 
-* On disks with an MBR partition table:
-  * If the OS is installed on a disk with an MBR partition table, and a
-    partition with the type id of 0xEA already exists it should be used as
-    `$BOOT`.
-  * Otherwise, if the OS is installed on a disk with an MBR partition table, a
-    new partition with type id of 0xEA shall be created, of a suitable size
-    (let's say 500MB), and it should be used as `$BOOT`.
 * On disks with GPT (GUID Partition Table)
-  * If the OS is installed on a disk with GPT, and an Extended Boot Loader
-    Partition (or XBOOTLDR partition for short), i.e. a partition with GPT type
-    GUID of `bc13c2ff-59e6-4262-a352-b275fd6f7172`, already exists, it should
+
+  * If an Extended Boot Loader Partition (XBOOTLDR partition for short) — a
```

Comment:
I don't like the wording of this. It sounds (to me) like XBOOT partitions replaces the ESP, when it really extends it. It should also be stated that only the ESP can contain the actual bootloader and its configuration (loader.conf etc.). The XBOOT can only old boot entries (type 1 or 2).

Reply:
Thanks for pointing this out. I reworked the text quite a bit, adding one more commit at the end.